src/NodeApi/JSReference.cs

```diff
@@ -192,14 +225,14 @@ protected virtual void Dispose(bool disposing)
 
             // The context may be null if the reference was created from a "no-context" scope such
             // as the native host. In that case the reference must be disposed from the JS thread.
-            if (SynchronizationContext == null)
+            if (_context == null)
             {
                 JSValueScope.CurrentRuntime.DeleteReference(_env, handle).ThrowIfFailed();
             }
             else
             {
-                SynchronizationContext.Post(
-                    () => JSValueScope.CurrentRuntime.DeleteReference(
+                _context.SynchronizationContext.Post(
+                    () => _context.Runtime.DeleteReference(
                         _env, handle).ThrowIfFailed(), allowSync: true);
```

Comment:
I wonder why we need the `handle` local variable?
Most probably it was added at some point to avoid capturing `this` in the lambda.
With the current use of other instance fields in the lambda, it may be not necessary anymore.
Though, we may look the other way around and avoid capturing `this`.

Reply:
I checked the history; at the time that variable was added, it was the only thing captured by the lambda. But I don't see any harm in capturing `this`; it might keep the object alive slightly longer, but that shouldn't be an issue. So I removed the extra variable.